### PR TITLE
Update coordinators admin interface

### PR DESCRIPTION
**Pull request by freesig**
### Summary
Adds updating of coordinator zomes to the admin interface.

### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/holochain_types/src/dna/coordinator_bundle.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+impl Manifest for CoordinatorManifest {
+    fn locations(&self) -> Vec<mr_bundle::Location> {
+        self.zomes
+            .iter()
+            .map(|zome| zome.location.clone())
+            .collect()
+    }
+
+    fn path() -> std::path::PathBuf {
+        "coords.yaml".into()
```
Comment (maackle) [inline crates/holochain_types/src/dna/coordinator_bundle.rs:35]: This seems like a misleading name, "coords" almost universally stands for "coordinates". 

Also does this have anything to do with bundling integrity and coordinator zomes together? Or is it just a bundle of coordinators?

Can we call it something boring like "zome-bundle"?

**Review by maackle — CHANGES_REQUESTED** [review]
Some changes, but otherwise seems fine.

I'm also wondering if there is an immediate use case for this, and if so, if there is any feedback from devs who would use this? Is this mainly for debugging?

**Comment by marycamacho** [comment]
Yikes ... anything named HOT swap in Holochain is misleading... many people looking at our open repo might assume the Holo HOT token swap when I'm sure that's not what was intended!

File: crates/holochain/tests/integrity_zome/mod.rs [hunk 2]
```diff
@@ -198,3 +211,58 @@ async fn test_coordinator_zome_hot_swap_multi_integrity() {
...
+    .unwrap()
+    .into();
+
+    let req = HotSwapCoordinatorsPayload {
+        dna_hash,
+        source: holochain_types::prelude::CoordinatorSource::Bundle(Box::new(source)),
+    };
+    let req = AdminRequest::HotSwapCoordinators(Box::new(req));
+    let r = admin_api.handle_admin_request(req).await;
+    assert!(matches!(r, AdminResponse::CoordinatorsHotSwapped));
```
[the remark this answers is not in the document]
Comment (freesig) [reply crates/holochain/tests/integrity_zome/mod.rs:267]: Can do. It is tested in the above tests by calling them and there's also tests on the dna itself but happy to add it here as well.

File: crates/holochain_types/src/dna/coordinator_bundle.rs (hunk 1 above)
Comment (freesig) [reply crates/holochain_types/src/dna/coordinator_bundle.rs:35]: It's just coordinators so I think `zome-bundle` would be misleading but I see the point about `coords`. Hmm, need another name for a set of coordinators. I'll have a think.

**Comment by freesig** [comment]
> Yikes ... anything named HOT swap in Holochain is misleading... many people looking at our open repo might assume the Holo HOT token swap when I'm sure that's not what was intended!

Ahh good point. I didn't think of that. I was just using the term like [this](https://en.wikipedia.org/wiki/Hot_swapping#Software). I'll try and think of a better name.

File: crates/holochain_types/src/dna/dna_file.rs [hunk 3]
```diff
@@ -87,11 +87,23 @@ impl DnaFile {
     }
 
     /// Hot swap coordinator zomes for this dna.
```
Comment (abe-njama) [inline crates/holochain_types/src/dna/dna_file.rs:89]: We still have code comments referencing Hot swap. Might want to change these to Update Coordinators as well. I saw some tests as well that have the term Hot swap in there.

File: crates/holochain_types/src/dna/dna_file/test.rs [hunk 4]
```diff
@@ -198,3 +198,92 @@ async fn test_hotswap() {
 
     assert_eq!(expect, dna);
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_hot_swap_checks_deps() {
```
Comment (maackle) [inline crates/holochain_types/src/dna/dna_file/test.rs:203]:
```suggestion
async fn test_update_coordinators_checks_deps() {
```

**Review by maackle — APPROVED** [review]
Looks good

**Pull request merged**